### PR TITLE
Add more Raspberry Pi Pico LCD Shields

### LINKED ISSUE #233: [HW] Waveshare Raspberry Pi Pico LCD Modules as Shields
**Is this request related to a missing driver support for a particular hardware platform, SoC or board? Please describe.**
I need support for Waveshare Pico LCD Modules as shields. The following shield should be supported:

- [x] Waveshare [Pico LCD 0.96](https://www.waveshare.com/wiki/Pico-LCD-0.96), 65K colors, 80 × 160 pixels, driver: [ST7735S](https://docs.zephyrproject.org/latest/build/dts/api/bindings/display/sitronix%2Cst7735r.html#dtbinding-sitronix-st7735r)
- [x] Waveshare [Pico LCD 1.3](https://www.waveshare.com/wiki/Pico-LCD-1.3), 65K colors, 240 × 240 pixels, driver: [ST7789VW](https://docs.zephyrproject.org/latest/build/dts/api/bindings/display/sitronix%2Cst7789v.html#dtbinding-sitronix-st7789v)
- [x] Waveshare [Pico LCD 1.44](https://www.waveshare.com/wiki/Pico-LCD-1.44), 65K colors, 128 × 128 pixels, driver: [ST7735S](https://docs.zephyrproject.org/latest/build/dts/api/bindings/display/sitronix%2Cst7735r.html#dtbinding-sitronix-st7735r)
- [x] Waveshare [Pico LCD 1.8](https://www.waveshare.com/wiki/Pico-LCD-1.8), 65K colors, 128 × 160 pixels, driver: [ST7735S](https://docs.zephyrproject.org/latest/build/dts/api/bindings/display/sitronix%2Cst7735r.html#dtbinding-sitronix-st7735r)
- [x] Waveshare [Pico-ResTouch-LCD-2.8](https://www.waveshare.com/wiki/Pico-ResTouch-LCD-2.8), 262K colors, 240 × 320 pixels, driver: [ST7789](https://docs.zephyrproject.org/latest/build/dts/api/bindings/display/sitronix%2Cst7789v.html#dtbinding-sitronix-st7789v) / [XPT2046](https://docs.zephyrproject.org/latest/build/dts/api/bindings/input/xptek%2Cxpt2046.html#dtbinding-xptek-xpt2046)

In this early first stage the Zephyr upstream samples "Shell on USB console", "LCD Orientation and Bit Order Test" and "LVGL Basic Sample" should be usable.

**Describe why you are asking for this support?**
With this support the board+shield variation for display based systems will scaling up.

I'm preparing a changeset for PR. I'll maintain the new board for Bridle and later contribute and maintain to/in Zephyr upstream.

**Additional context**
The new shields support will coming together with RPi Pico R3 connector support as currently provided by Bridle only.

### PR DESCRIPTION
**Pull request by rexut**
Fixes #233 and also brings many improvements in shield documentation:

- Pico LCD 0.96 shield by Waveshare
- Pico LCD 1.3 shield by Waveshare
- Pico LCD 1.44 shield by Waveshare
- Pico LCD 1.8 shield by Waveshare
- Pico ResTouch LCD 2.8 shield by Waveshare

### CODE REVIEW

**Comment by rexut** [comment]
> You can find the documentation preview for this PR at this [link](https://bridle.tiac-systems.net/ghpr/doc/PR-238 ). It will be updated about 10 minutes after the documentation build succeeds.
> 
> _Note: This comment is automatically posted by the Documentation Publishing GitHub Action._

Found the new shields here: https://bridle.tiac-systems.net/ghpr/doc/PR-238/bridle/boards/shields/rpi_pico_lcd/doc/index.html

File: boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_44/positions.rsti [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          :width: 500
+          :alt: Waveshare Pico LCD 1.44 details
+
+   * - .. container:: twocol
+
+          .. container:: leftside
+
+             Inputs:
+
+             1. | :strong:`ST7735S`
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_44/positions.rsti:17]: **praise (blocking):** no input – resort the numbers

File: boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_8/positions.rsti [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          :width: 500
+          :alt: Waveshare Pico LCD 1.8 details
+
+   * - .. container:: twocol
+
+          .. container:: leftside
+
+             Inputs:
+
+             1. | :strong:`ST7735S`
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_8/positions.rsti:17]: **praise (blocking):** no input – resort the numbers

File: boards/shields/rpi_pico_lcd/boards/waveshare_pico_lcd_0_96/rpipico_r3.dtsi [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+&rpipico_spi_lcd {
+	pinctrl-0 = <&spi1_lcd>;
+	pinctrl-names = "default";
+	cs-gpios = <&rpipico_header 9 GPIO_ACTIVE_LOW>;		/* GP9 */
+};
+
+&mipi_dbi {
+	status = "okay";
+	reset-gpios = <&rpipico_header 12 GPIO_ACTIVE_LOW>;	/* GP12 */
+	dc-gpios = <&rpipico_header 8 GPIO_ACTIVE_HIGH>;        /* GP8 */
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/boards/waveshare_pico_lcd_0_96/rpipico_r3.dtsi:20]: **praise (blocking):** use tabulator before comment `/* GP8 */` – check also all other new shields

File: boards/shields/rpi_pico_lcd/doc/index.rst [hunk 4]
```diff
@@ -758,6 +840,88 @@ sample: :zephyr:code-sample:`zephyr:input-dump`.
          I: input event: dev=gpio_keys        SYN type= 1 code= 28 value=1
          I: input event: dev=gpio_keys        SYN type= 1 code= 28 value=0
 
+   .. group-tab:: Waveshare Pico LCD 1.44
+
+      Print the input events related to the five on-shield joystick keys
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/doc/index.rst]: **praise (blocking):** fix description, there is no joystick on the Waveshare Pico LCD 1.44 – check / verify also all other new shield descriptions

File: boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_8/pinouts.rsti [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+       - | :rpi-pico-pin:`38` :rpi-pico-gnd:`GND`
+       - | :rpi-pico-pin:`39` :rpi-pico-vdd:`VSYS`
+         | :rpi-pico-pin-nc:`nc` on-shield LDO power enable
+         | :rpi-pico-pin-nc:`nc` (default) :rpi-pico-sys:`VCC_EN`
+       - | :rpi-pico-pin:`40` :rpi-pico-sys:`VBUS`
+
+       .. rubric:: Devicetree compatible
+
+       - :dtcompatible:`raspberrypi,pico-header-r3`
+       - :dtcompatible:`gpio-keys`
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_8/pinouts.rsti:84]: **praise (blocking):** remove link to `gpio-keys`, this shield has no input elements

File: boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_44/positions.rsti (hunk 1 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_44/positions.rsti]: done

File: boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_8/positions.rsti (hunk 2 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_8/positions.rsti:15]: done

File: boards/shields/rpi_pico_lcd/boards/waveshare_pico_lcd_0_96/rpipico_r3.dtsi (hunk 3 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/boards/waveshare_pico_lcd_0_96/rpipico_r3.dtsi]: done

File: boards/shields/rpi_pico_lcd/doc/index.rst (hunk 4 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/doc/index.rst]: done

File: boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_8/pinouts.rsti (hunk 5 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_8/pinouts.rsti]: done

**Comment by rexut** [comment]
**CONVERTED TO DRAFT** see review commits in PR #243

File: boards/shields/rpi_pico_lcd/doc/index.rst [hunk 6]
```diff
@@ -2730,497 +366,59 @@ ResTouch LCD 3.5| board. They will be built with activated USB-CDC/ACM console.
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/doc/index.rst:366]: **praise (blocking):** Change paragraph to: _This samples and test applications aren't applicable on all boards. They will be built with activated USB-CDC/ACM console._

File: boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_8/lvgl_basic.rsti [hunk 7]
```diff
@@ -0,0 +1,81 @@
+Using the LVGL module on top of the :ref:`Display driver API
+<zephyr:display_api>` with chosen display. That is:
+
+| :hwftlbl-scr:`LCD` : :devicetree:`chosen { zephyr,display = &lcd_panel; };`
+| :hwftlbl-scr:`ST7789V` : :devicetree:`lcd_panel: &st7735s_160x128 {};`
```
Comment (rexut) [inline boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_8/lvgl_basic.rsti:5]: **praise (blocking):** change to `ST7735S`

File: boards/shields/rpi_pico_lcd/doc/index.rst (hunk 6 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/doc/index.rst:366]: done

File: boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_8/lvgl_basic.rsti (hunk 7 above)
Comment (rexut) [reply boards/shields/rpi_pico_lcd/doc/waveshare_pico_lcd_1_8/lvgl_basic.rsti]: done

**Comment by rexut** [comment]
> **CONVERTED TO DRAFT** see review comments in PR #243

done –> **READY FOR REVIEW**

**Review by tobiaskaestner — DISMISSED** [review]
LGTM

**Review by tobiaskaestner — APPROVED** [review]
Lgtm

**Pull request merged**